Assets/Scripts/Enemy.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Enemy : MonoBehaviour
+{
+    //component
+    Rigidbody2D rigid;
+    SpriteRenderer spriteRenderer;
+    Animator anima;
+    [SerializeField] GameObject target;
+    [SerializeField] private int maxHP;
+    public static int currentHP;
+    //movement direction related
+    [SerializeField] private int nextMove;
+    private bool notMoving;
+    private bool moveUp;
+    private bool moveDown;
+    private bool moveRight;
+    private bool moveLeft;
+    //chasing player related
+    [SerializeField] private bool chaser = false;
+    [SerializeField] private bool isChasing;
+
+    private void Awake()
+    {
+        currentHP = maxHP;
+        rigid = GetComponent<Rigidbody2D>();
+        spriteRenderer = GetComponent<SpriteRenderer>();
+        anima = GetComponent<Animator>();
+        StartCoroutine("ChangeMovement");
+    }
+
+    private void Update()
+    {
+        //death condition
+        if (currentHP <= 0)
+        {
+            Destroy(gameObject);
+        }
+
+        //movement direction
+        switch (nextMove)
+        {
+            case 0:

```

Comment:
This is a very long switch case that could maybe be reduced by changing each bool that checks whether the enemy is moving into a Vector 2 that simply states the movement vector. For example: NotMoving = new Vector 2(0,0) and when case 0 is selected you invoke NotMoving, or MovingRight = new Vector2(1,0) and when case 4 is chosen, invoke MovingRight. It's a bit hard to understand with all the booleans. Also instead of having nextMove as an int, maybe have it as an Enum so that you can check which state is happening (Up, Down, Left, Right) this will also save some lines as it'll remove the Move() method